src/java/org/apache/cassandra/repair/state/CoordinatorState.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.repair.state;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import org.apache.cassandra.db.ColumnFamilyStore;
+import org.apache.cassandra.locator.InetAddressAndPort;
+import org.apache.cassandra.repair.CommonRange;
+import org.apache.cassandra.repair.RepairRunnable;
+import org.apache.cassandra.repair.messages.RepairOption;
+import org.apache.cassandra.utils.UUIDGen;
+
+public class CoordinatorState extends AbstractState<CoordinatorState.State>
+{
+    public enum State
+    {
+        INIT, SETUP, STARTED,
+        PREPARE_SUBMIT, PREPARE_COMPLETE,
+        REPAIR_SUBMIT, REPAIR_COMPLETE,
+        SKIPPED, SUCCESS, FAILURE;
+
+        public static final int NUM_STATES = values().length - 2; // 3 represent COMPLETE, so remove duplicate
+    }
+
+    public final UUID id = UUIDGen.getTimeUUID();
```

Comment:
is there a reason for another id? `parentSessionid` should be able to uniquely identify coordinator state

Reply:
this is `parentSessionid`, that got moved here and named `id`